### ISSUE TITLE
General feedback

### ISSUE THREAD

**Issue by ealmansi** [rincemust/LongNums#3]
There's a few other points I thought of bringing up when I was doing the minor refactoring in my last PR. I haven't really had time to suggest these via a PR, so I'll just write the list of comments here:

- The name of the class would probably feel more intuitive in singular and with consistent case. Ie: `LongNumber` or `BigNumber` or `BigInteger` instead of `Longnums`.
- You're defining a custom type supporting "natural" operations via operator overloading, yet it does not conform to the typical interface. Here's a really good reference as to which operators your custom type would be expected to implement: http://articles.emptycrate.com/2009/10/12/nobody_understands_c_part_8_operator_overloading.html.
- There's no need to define new operators for each int type you want your custom type to be interoperable with. For example, if you want your custom type to be able to perform arithmetic operations, or be comparable with, long ints simply add a constructor with the signature `Longnums(const long int x)` and you get all these operations for free.
- Functions returning a boolean value derived from an expression should return the expression itself. I mean doing this `return expr` instead of this `if (expr) return true; else return false;`.
- You have duplicated logic in quite a few places. For example, when implementing comparison operators you would want to define only two of them (`==` and `<`, for example), and define the rest in terms of these two. Also, instead of implementing addition and subtraction you could define subtraction as `a - b = a + (-b)`.
- I'm not sure why you're adding a `base` property to the Longnums class, since this is expected to be always the same for all instances of the class. Not sure if I'm missing something here, tho.
- A better alternative to using base 10^8 would probably be just using unsigned ints to represent your numbers (essentially using base 2^32).

Best of luck. Cheers.

**Comment by rincemust** [comment]
Thanks a lot! I'll try to implement these as and when I can. As for the base, I had used 10^8 to just to make it easier to implement the logic for the multi-word numbers. Now it is just a matter of changing the value of `base`. I am also planning to change the `base` variable from being a member of the class to just being a variable used to define the base in the `.cpp` file, since it is unnecessary as of now.

**Comment by ealmansi** [comment]
Cool. CC me if you'd like some peer review once you submit the changes and/or further tests.

**Comment by aadityanaik** [comment]
So, I've implemented most of the fixes you suggested. As for the base of the numbers, I have assigned it the closest power of 10 less than the value of `INT_MAX`. I have not yet figured out a good method to use 2^32 as the base, but I will work on it. I also added the bash script which I used to test the outputs. It makes it easier to test for longer integers. Any more suggestions would be welcome!
